### PR TITLE
Deploy to S3 via Travis

### PR DESCRIPTION
**Pull request by jaebradley**
image

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
 rvm:
-  - 2.3.3
-  - 2.4.0
-
+- 2.4.5
 cache: bundler
 script: bundle exec middleman build
+deploy:
+  provider: s3
+  access_key_id: $AWS_ACCESS_KEY_ID
```
Comment (jaebradley) [inline .travis.yml:9]: I'm just using Travis encrypted variables in "Settings"

image

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+  secret_access_key: $AWS_SECRET_ACCESS_KEY
+  bucket: developer.grooveapp.com
+  local-dir: build
+  upload-dir: documentation
```
Comment (jaebradley) [inline .travis.yml:13]: I'm uploading this stuff to a `documentation` dir and then the default behavior is to point to this directory.

Maybe this is cleaner than uploading all the files to the bucket directly?

File: .travis.yml (hunk 1 above)
Comment (vijaypemmaraju) [reply .travis.yml:9]: ah cool, I think I actually like this better than using travis's in-file secrets

File: .ruby-version [hunk 3]
```diff
@@ -1 +0,0 @@
-2.4.5
```
Comment (jaebradley) [inline .ruby-version]: Lemme keep this

File: .travis.yml [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
+  skip_cleanup: true
+  region: us-west-2
+  acl: public_read
+  on:
+    all_branches: true
+    condition: $TRAVIS_BRANCH = "master"
+  after_deploy:
+  - echo "*** This build was deployed to $PUBLIC_URL"
```
Comment (vijaypemmaraju) [inline .travis.yml]: probably don't need this, right?

**Review by vijaypemmaraju — APPROVED** [review]

**Pull request merged**